### ISSUE TITLE
[FEATURE REQ] Upgrade to spring-boot 3.1.x

### ISSUE THREAD

**Issue by davidkarlsen** [Azure/azure-sdk-for-java#35065]
**Is your feature request related to a problem? Please describe.**
Want to use latest spring-boot libraries.

**Describe the solution you'd like**
Make the https://repo1.maven.org/maven2/com/azure/spring/spring-cloud-azure-dependencies/ compatible with SB3.1

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Description Added
- [X Expected solution specified

**Comment by blackr1234** [comment]
Spring Boot 3.1.0 is just officially released.
Spring Boot 3.0.x will be end of life in 6 months.

May I know the plan or tentative release date of a version that supports Spring Boot 3.1?
Thanks. 🙏

**Comment by Netyyyy** [comment]
Hi, @davidkarlsen and @blackr1234 Thanks for reaching out. 
We will support 3.1.x, but doesn't have an exact time yet.

**Comment by boskokg** [comment]
Please do this asap. this is the only thing that blocks us to upgrade spring boot and fix bugs that we have now.

**Comment by juliojgd** [comment]
Yes, please, take it ASAP @Netyyyy . It's blocking our upgrade as well, we though this was prepared in advance as Microsoft and Spring team is working closely to improve support for Spring applications in Azure deployments...

**Comment by apahne** [comment]
Yes, that would be highly appreciated.

**Comment by riao78** [comment]
desperately waiting for this

**Comment by Netyyyy** [comment]
Hi all, we are doing a test now, any progress will be updated here, thanks

**Comment by Heatmanofurioso** [comment]
Same here.
Many projects are eagerly updating to 3.1.0

**Comment by boskokg** [comment]
@Netyyyy do you have any update on this? Can we expect the release soon?

**Comment by Netyyyy** [comment]
Sorry for the delayed response, we're still working on it, 
there are some problems when upgrading to Jackson 2.15.0, trying to fix them.

**Comment by marccollin** [comment]
ms is slow, spring boot is major in java world

**Comment by ikannak** [comment]
For those who are being blocked by this upgrade, I have found that `spring.cloud.azure.compatibility-verifier.enabled=false` in your application.properties disables the version check. After adding this property and updating my Spring Cloud for Azure dependencies to 5.2.0 I can use the features of Spring Cloud Azure that I use (Mainly using Azure Active Directory for logging in). Spring Boot 3.1 is a minor version increase and AFAIK doesn't have any major breaking changes.

**Comment by juliojgd** [comment]
> For those who are being blocked by this upgrade, I have found that `spring.cloud.azure.compatibility-verifier.enabled=false` 

Thanks @ikannak , we were aware of that and we are using it as workaround, but we need proper Spring Boot 3.1.x support, not a fragile workaround like this 😄 .

**Comment by blackr1234** [comment]
@ikannak 
Thanks for bringing up the version check property but I think everyone already knows it as this is mentioned in the error message when starting the Spring Boot app. Also, we cannot assume that minor version upgrades won't break things.
I think the whole point of this ticket is to request Azure team to provide an Azure Java SDK version that is "officially compatible" with the latest release version of the Spring Boot framework.
I agree with @juliojgd on his point about "fragile workaround". In large companies, we are not allowed to use non-production-ready SDKs or incompatible SDKs or SDKs that are not officially endorsed.
Therefore, unless the official [Spring Versions Mapping](https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Versions-Mapping) page clearly states that there is an Azure Java SDK version compatible with Spring Boot 3.1, we have to wait for Microsoft to release or announce it.

---

@Netyyyy 
The support from your team is greatly appreciated.
As a week has passed, may I know if there is any update and any delivery plan? Thank you.

**Comment by Sti2nd** [comment]
Isn't this more like a bug than a feature request. Third-party libraries shouldn't prevent minor upgrade of a framework (or anything)?

**Comment by boskokg** [comment]
@Netyyyy can you please update us with the progress?
Is this issue wiith a low priority for your team or you are stuck with the issue?
Best regards

**Comment by saragluna** [comment]
> Thanks for the update.
> 
> Is there an ETA on when the Spring Cloud Azure section on Spring Security Support (https://learn.microsoft.com/en-us/azure/developer/java/spring-framework/spring-security-support?tabs=SpringCloudAzure5x) will be updated with guidance on how to migrate to the lambda based security method, so without using and()? It appears the examples all still use .apply(AadWebApplicationHttpSecurityConfigurer.aadWebApplication()).and(), which causes a deprecation warning.

Sorry but no ETA right now, PR is welcome.
